packages/plugins/src/DocumentAssetsPlugin.ts

```diff
@@ -0,0 +1,174 @@
+import type { Page, Plugin as PluginType } from '@jpmorganchase/mosaic-types';
+import fsExtra from 'fs-extra';
+import glob from 'fast-glob';
+import path from 'path';
+import { escapeRegExp } from 'lodash-es';
+import { unified } from 'unified';
+import { visit } from 'unist-util-visit';
+import remarkParse from 'remark-parse';
+import remarkStringify from 'remark-stringify';
+import { VFile } from 'vfile';
+
+interface DocumentAssetsPluginOptions {
+  /**
+   * A collection of subdirectory names that contain assets, referenced by pages in the associated parent directory
+   * @default: ['images']
+   */
+  assetSubDirs?: string[];
+  /**
+   * The source path, where your docs reside, when the site runs
+   */
+  srcDir?: string;
+  /**
+   * The directory to copy matched assets to, typically the site's public directory
+   * @default './public'
+   */
+  outputDir?: string;
+  /**
+   * The prefix we add to all images in documents, so that it routes to the public directory
+   * @default '/images'
+   */
+  imagesPrefix?: string;
+}
+
+function isUrl(assetPath: string): boolean {
+  try {
+    new URL(assetPath);
+    return true;
+  } catch (_err) {}
+  return false;
+}
+
+function isRelativePath(assetPath: string): boolean {
+  const relativePathRegExp = /^(\.{1,2}\/)+/;
```

Comment:
Is it worth highlighting paths are expected to be posix pathing?

Or the regex can be updated:

```suggestion
  const relativePathRegExp = /^(\.{1,2}(\/|\\\\))+/;
```

Reply:
refactored and removed this function